src/core/mac/mac_frame.cpp

```diff
@@ -975,6 +974,8 @@ const uint8_t *Frame::GetHeaderIe(uint8_t aIeId) const
     return header;
 }
 
+#endif // OPENTHREAD_CONFIG_MAC_HEADER_IE_SUPPORT
```

Comment:
Notice this `#endif` can be removed together with the `#if` around line 897, which is redundant.